Blog-Website/server/controller/user-controller.js

```diff
@@ -14,7 +14,12 @@ export const singupUser = async (request, response) => {
         const hashedPassword = await bcrypt.hash(request.body.password, 10);
 
         const user = { username: request.body.username, name: request.body.name, password: hashedPassword }
-
+         //we need to first check if user already present or not if present then we have to throw an error otherwise we have
```

Comment:
We should move these lines above the password hashing step because if the username already exists, there's no benefit in hashing the password if we're not going to save the user details in the database.